docs/specifications/tests/Nameserver-TP/nameserver02.md

```diff
@@ -1,61 +1,113 @@
 > defined in this document MUST respond with a return code (RCODE) of
 > FORMERR to messages containing an OPT record in the additional
 > section and MUST NOT include an OPT record in the response.
 
-### Inputs
-
-The domain name to be tested.
+Servers supporting EDNS(0) must reply with EDNS(0)
+([RFC 6891, section 6.1.1]):
 
-### Ordered description of steps to be taken to execute the test case
-
-1. Retrieve all address records for all the name servers using
-   [Method 4](../Methods.md#method-4-obtain-glue-address-records-from-parent) and
-   [Method 5](../Methods.md#method-5-obtain-the-name-server-address-records-from-child),
-   and do recursive lookups for the name servers that are out of bailiwick.
-2. Send a DNS query to each name server IP address querying the SOA record
-   of the domain name with EDNS0 option of payload size ("bufsize")
-   set to 512.
-3. If any answer from step 2 contains a FORMERR RCODE this test case fails.
-4. If the answer does not contain an OPT RR with EDNS version 0, this test
-   case fails.
+> If an OPT record is present in a received request, compliant
+> responders MUST include an OPT record in their respective responses.
 
-### Outcome(s)
 
-If any name server returns FORMERR for a query using EDNS(0), or if there is
-no OPT RR with EDNS version 0 in it, this test case fails.
+## Inputs
 
-### Special procedural requirements
+* "Child Zone" - The domain name to be tested.
 
-None.
+## Ordered description of steps to be taken to execute the test case
 
-### Intercase dependencies
-
-None.
-
--------
+1. Created an SOA query for the *Child Zone* with an OPT record with 
+   EDNS version set to "0" and with EDNS(0) option of payload size ("bufsize")
+   set to 512.
 
-Copyright (c) 2013, 2014, 2015, IIS (The Internet Infrastructure Foundation)  
-Copyright (c) 2013, 2014, 2015, AFNIC  
-Creative Commons Attribution 4.0 International License
+2. Create a second SOA query for the *Child Zone* without any OPT record.
+
+3. Obtain the set of name server IP addresses using [Method4] and [Method5]
+   ("Name Server IP").
+
+4. For each name server in *Name Server IP* do:
+
+   1. Send the SOA query **with** OPT record to the name server and collect 
+      the response.
+   2. If there is no DNS response, then:
+      1. Send the SOA query **without** OPT record to the name server and 
+         collect the response.
+      2. If there is no DNS response, then output *[NO_RESPONSE]* and 
+         go to next server.
+      3. Else (there is a DNS response), then output 
+         *[BROKEN_EDNS_SUPPORT]* and go to next server.
+   3. Else, if the DNS response meet the following two criteria,
+      then output *[NO_EDNS_SUPPORT]*:
+      1. It has the RCODE "FORMERR" 
+      2. It has no OPT record.
+   4. Else, if the DNS response meet the following two criteria,
+      then just go to the next name server (compliant server):
+      1. It has the RCODE "NOERROR".
+      2. It has OPT record with EDNS version 0.
```

Comment:
3. The answer section contains the SOA record

Reply:
Done.